unitary/examples/quantum_rpg/battle.py

```diff
@@ -0,0 +1,120 @@
+            for testing.
+    """
+
+    def __init__(self,
+                 player_side: List[Qaracter],
+                 enemy_side: List[Qaracter],
+                 file: io.IOBase = sys.stdout):
+        self.player_side = player_side
+        self.enemy_side = enemy_side
+        self.file = file
+
+    def print_screen(self):
+        """Prints a two-column output of the battle status.
+
+        Left side includes the players and their qubits.  Right
+        side includes the NPCs and their qubits.
+
+        Output will be written to the `file` attribute.
+        """
+        print('-----------------------------------------------', file=self.file)
+        for i in range(max(len(self.player_side), len(self.enemy_side))):
+            status = ''
+            if i < len(self.player_side):
+                status += f'{self.player_side[i].name} {type(self.player_side[i]).__name__}'
+            else:
+                status += '\t\t'
+            status += '\t\t\t'
+            if i < len(self.enemy_side):
+                status += f'{self.enemy_side[i].name} {type(self.enemy_side[i]).__name__}'
+
+            status += '\n'
+
+            if i < len(self.player_side):
+                status += self.player_side[i].status_line()
+            else:
+                status += '\t\t'
+            status += '\t\t\t'
+            if i < len(self.enemy_side):
+                status += self.enemy_side[i].status_line()
+            print(status, file=self.file)
+        print('-----------------------------------------------', file=self.file)
+
+    def take_player_turn(self, user_input: Optional[List[str]] = None):
+        """Take a player's turn and record results in the battle.
+
+        1) Retrieve the possible actions from the player.
+        2) Prompt the player for which action to use.
+        3) Prompt the player which NPC and qubit to target.
+        4) Call the player's action to perform the action.
+
+        Args:
+            user_input: List of strings that substitute for the user's
+                raw input.
+        """
+
+        # If user input is provided as an argument, then use that.
+        # Otherwise, prompt from raw input.
+        if user_input is not None:
+            user_input = iter(user_input)
+            get_user_input = lambda _: next(user_input)
+        else:
+            get_user_input = input
+
+        for current_player in self.player_side:
+            self.print_screen()
+            print(f'{current_player.name} turn:', file=self.file)
+            if not current_player.is_active():
+                print(f'{current_player.name} is DOWN!', file=self.file)
+                continue
+            actions = current_player.actions()
+            for key in actions:
+                print(key, file=self.file)
+            action = get_user_input('Choose your action: ')
+            if action in current_player.actions():
+                monster = int( get_user_input('Which enemy number: ')) + 1
```

Comment:
Shouldn't this be subtracting 1 from the input?

Reply:
Good catch!  Fixed and added more unit tests for this case.